### PR TITLE
src-userProfiles list view and details view

### LINKED ISSUE #33: View User Profile Details
As an admin, I would like to see the content of a User Profile so I can read it.

**Given** a user is viewing a list of User Profiles  
**When** they select a user
**Then** they should be directed to a User Profile Detail page

User Profile Details include:

* Full name
* Avatar image (if exists, else use a default image)
* Display name
* Email
* Creation Date (MM/DD/YYYY)
* User Profile type

### LINKED ISSUE #34: View All User Profiles
As an admin, I would like to see a list of all the User Profiles so that I can keep track of who is using the system.

**Given** the user is an admin in the Tabloid application  
**When** they select the `User Profiles` menu option  
**Then** they should be directed to the User Profiles list page  
**And** each user in the list should display the full name, the display name and the user type.
**And** the list should be in ordered alphabetically by user display name

### PR DESCRIPTION
**Pull request by sorachung**
#### Changes Made
1. Added file `quill-logo.png` to `client/src/images/` directory.
2. Added file `UserProfileManager.js` to `client/src/modules/` directory. Includes fetch calls - `getAllUserProfiles` and `getUserProfile`.
3. Added file `UserProfilesList.js` to `client/src/components/` directory. Includes view of a list of all active user profiles.
4. Added file `UserProfileListItem.js` to `client/src/components/` directory. Includes a single list item of a profile to be used in `UserProfilesList.js`
5. Added file `UserProfileDetails.js` to `client/src/components/` directory. Includes all relevant details of a single user.

1. Modified file `UserProfileRepository.cs and IUserProfileRepository.cs` to include `GetById and GetAll` methods and functionality.
2. Modified file `UserProfileController.cs` to include `Get() and Get(int id)` methods and functionality.
6. Modified file `UserProfile.cs` to include `CreateDateTimeFormatted` computed property.
7. Modified file `ApplicationViews.js` to include `Routing to User Profiles List and Details and 404 `.
​
#### Steps to Review
1. Checkout this branch locally.
    ```
    git fetch --all
    git checkout src-userProfiles
    ```
8. Open a new Terminal tab (⌘T) and navigate to the server directory.
9. Test app functionality.
    > Log in as a user
    > click `User Profiles` in the header
    > verify that all of the users show up in order of display name (alphabetically)
    > click details of a few users
    > verify that all of the info as required in issue ticket #33 shows
    > when you go to a url of something like `/userprofiles/details/10000` aka the detail page of a user that does not exist, you should see a "404 not found".
10. View code file.
    > Confirm file modifications are present as indicated above.
    > Confirm no unused code or extraneous comments exist.

closes #34 
closes #33

### CODE REVIEW

**Comment by aliparadise** [comment]
😊

**Comment by heymonicakay** [comment]
This pull request includes changes related to more than one user story, and it includes changes unrelated to any open issue tickets as well (PR template, 404 not found). On the job, you will not be allowed to submit this large of a pull request at once. Additionally, you'll want to open a new issue ticket for enhancements like 404 routing, and contribution requirements (pr templates).

File: PULL_REQUEST_tEMPLATE.md [hunk 1]
```diff
@@ -0,0 +1,19 @@
+```text
```
[the remark this answers is not in the document]
Comment (sorachung) [reply PULL_REQUEST_tEMPLATE.md]: done!

**Review by heymonicakay — APPROVED** [review]

File: Tabloid/Repositories/UserProfileRepository.cs [hunk 2]
```diff
@@ -55,6 +56,104 @@ public UserProfile GetByFirebaseUserId(string firebaseUserId)
             }
         }
 
+        public List<UserProfile> GetAll()
```
[the remark this answers is not in the document]
Comment (sorachung) [reply Tabloid/Repositories/UserProfileRepository.cs:59]: We actually went over this use of DBUtils earlier today. I touched base with them just now to make sure that everyone knew what was up though.

**Pull request merged**